Update cloud provider modules

tklauser opened the pull request:
Monthly update of the cloud provider SDK Go modules. Also add a script to simplify the update.

tklauser committed:
contrib/scripts: add script to update cloud provider SDKs

Add a script for the monthly update of the cloud provider SDK Go
modules.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

aditighag commented on contrib/scripts/go-mod-update-cloud-providers.sh:
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
```
nit : Doesn't look like this is needed?

tklauser replied:
It's currently not needed, but most other scripts have this by default and I think it's good practice to already set these in case of future changes introducing piped commands.

aditighag reviewed (changes requested):
Thanks for adding the script. 🚀 Can we document the script?

tklauser committed:
go.mod, vendor: update cloud SDK modules

Update the cloud SDK modules now using the newly added
contrib/scripts/go-mod-update-cloud-providers.sh script.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
